alembic/env.py

```diff
@@ -1,76 +1,109 @@
 from __future__ import with_statement
-from alembic import context
-from sqlalchemy import engine_from_config, pool
+
+import logging
+import sys
 from logging.config import fileConfig
+from alembic import context
 
-import os, sys
-sys.path.insert(0, os.path.realpath(os.path.join(os.path.dirname(__file__), '..')))
+import sideboard
+import uber
+from sideboard.lib.sa import CoerceUTF8, UTCDateTime, UUID
+from uber.migration import version_locations_option
+from uber.models import Choice, MultiChoice, Session
 
-# this is the Alembic Config object, which provides
-# access to the values within the .ini file in use.
-config = context.config
 
-# Interpret the config file for Python logging.
-# This line sets up loggers basically.
-fileConfig(config.config_file_name)
+# This is the alembic Config object, which provides access to "alembic.ini".
+alembic_config = context.config
 
-# add your model's MetaData object here
-# for 'autogenerate' support
-# from myapp import mymodel
-# target_metadata = mymodel.Base.metadata
-import sideboard
-from uber.common import *
+version_locations = alembic_config.get_main_option('version_locations')
+assert version_locations == version_locations_option, (
+        'alembic must be run using the sep command:\n'
+        'sep alembic {}'.format(' '.join(sys.argv[1:])))
+
+# Interpret the config file for Python logging. This sets up alembic loggers.
+if alembic_config.config_file_name:
+    fileConfig(alembic_config.config_file_name)
+logger = logging.getLogger('alembic.env')
+
+# Add the model's MetaData object here for "autogenerate" support.
 target_metadata = Session.BaseClass.metadata
 
-# other values from the config, defined by the needs of env.py,
-# can be acquired:
-# my_important_option = config.get_main_option("my_important_option")
-# ... etc.
-# We replace the INI option with RAMS' URL.
-config.set_main_option('sqlalchemy.url', c.SQLALCHEMY_URL)
+
+def include_object(object, name, type_, reflected, compare_to):
+    """Exclude alembic's own version tables from alembic's consideration."""
+    return not name.startswith('alembic_version')
+
+
+def render_item(type_, obj, autogen_context):
+    """Apply custom rendering for selected items."""
+    if type_ == 'type':
+        if isinstance(obj, Choice):
+            return 'sa.Integer()'
+        elif isinstance(obj, UTCDateTime):
+            autogen_context.imports.add('import sideboard')
+            return 'sideboard.lib.sa.UTCDateTime()'
```

Comment:
Very slight nitpick, but saying ``import sideboard`` doesn't automatically make ``sideboard.lib.sa`` available:

```python
(env)eli@haku:~/sideboard$ python
Python 2.7.10 (default, Oct 14 2015, 16:09:02) 
[GCC 5.2.1 20151010] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sideboard
>>> sideboard.lib.sa.UTCDateTime
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'sa'
```

Since you've tested this and found that it actually works, then I'm guessing that you happen to be importing ``sideboard.lib.sa`` elsewhere or something.  But I believe it would be more correct to say

```python
autogen_context.imports.add('import sideboard.lib.sa')
```

here, unless I've misunderstood something.